### PR TITLE
Add news crawler

### PR DESCRIPTION
**Pull request by mahalrs**
Implemented news crawler. It downloads html page content which can be used to extract image and image captions.

To run it locally:
```
cd news_crawler/crawler
npm install
npm run build

# You need to provide it a crawl-input.json file that contains crawl config,
# target domains, starting urls...
npm run start -- --crawl-config ./crawl-input.json --dir-path ./data
```

### CODE REVIEW

File: news_crawler/crawler/src/index.ts
Comment (ArjunKrishna3367) [inline news_crawler/crawler/src/index.ts:1]: Could you give more details on how the crawl input json file should be formatted (or even better, include an example json that is already formatted)?

**Review by ArjunKrishna3367 — COMMENTED** [review]
Could you give more details on how the crawl input json file should be formatted (or even better, include an example json that is already formatted)?

**Comment by mahalrs** [comment]
> Could you give more details on how the crawl input json file should be formatted (or even better, include an example json that is already formatted)?

Here's a sample json input. Save it as `crawl-input.json`.

`targetWebsites` is a dictionary of target domains as the key and value is the target paths to crawl for each domain. For example, TechCrunch articles has path like '/2023/...' but non-article pages on TechCrunch has different path.

`startUrls` is a list of links to start with (e.g. TechCrunch homepage to get scrape article links).

```
{
    "crawlName": "local-crawl",
    "crawlId": "test1",
    "targetWebsites": {
        "techcrunch.com": [
            "/20"
        ]
    },
    "startUrls": [
        "https://techcrunch.com/"
    ]
}

```

**Review by ArjunKrishna3367 — CHANGES_REQUESTED** [review]
Looks good, I was able to get it running now. The comments in the code were helpful.
Just need to create a README file with the two comments that you posted in the PR (the description of how to run it and the sample json), then should be good for merging.

**Comment by mahalrs** [comment]
> Looks good, I was able to get it running now. The comments in the code were helpful. Just need to create a README file with the two comments that you posted in the PR (the description of how to run it and the sample json), then should be good for merging.

@ArjunKrishna3367 Done

**Review by ArjunKrishna3367 — APPROVED** [review]

**Pull request merged**